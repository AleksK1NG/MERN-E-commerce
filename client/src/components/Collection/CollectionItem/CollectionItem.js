import React from 'react'
import { connect } from 'react-redux'
import { addCartItem } from '../../../storeModules/cart/cartActions'

import {
  CollectionFooterContainer,
  AddButton,
  BackgroundImage,
  CollectionItemContainer,
  NameContainer,
  PriceContainer
} from './CollectionItem.styles'

const CollectionItem = ({ item, addCartItem }) => {
  const { name, price, imageUrl } = item

  return (
    <CollectionItemContainer>
      <BackgroundImage className="image" imageUrl={imageUrl} />

      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
      </CollectionFooterContainer>

      <AddButton onClick={() => addCartItem(item)} inverted>
        Add to cart
      </AddButton>
    </CollectionItemContainer>
  )
}

export default connect(
  null,
  { addCartItem }
)(CollectionItem)
