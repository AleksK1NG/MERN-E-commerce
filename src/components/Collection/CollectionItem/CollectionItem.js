import React from 'react'
import './CollectionItem.styles.scss'
import CustomButton from '../../Shared/CustomButton/CustomButton'

const CollectionItem = ({ id, name, price, imageUrl }) => {
  return (
    <div className="collection-item">
      <div style={{ backgroundImage: `url(${imageUrl})` }} className="image" />

      <div className="collection-footer">
        <span className="name">{name}</span>
        <span className="price">${price}</span>
      </div>

      <CustomButton inverted>Add to CART</CustomButton>
    </div>
  )
}

export default React.memo(CollectionItem)
