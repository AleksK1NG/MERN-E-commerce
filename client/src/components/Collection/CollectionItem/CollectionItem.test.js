import React from 'react'
import { shallow } from 'enzyme'
import { CollectionItem } from './CollectionItem'

describe('Test CollectionItem component', () => {
  let wrapper
  let mockAddCartItem

  const mockItem = { name: 'black hat', price: 250, imageUrl: 'www.testImage.com' }

  beforeEach(() => {
    mockAddCartItem = jest.fn()

    const mockProps = {
      item: mockItem,
      addCartItem: mockAddCartItem
    }

    wrapper = shallow(<CollectionItem {...mockProps} />)
  })

  it('should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call addItem when AddButton clicked', () => {
    wrapper.find('AddButton').simulate('click')
    expect(mockAddCartItem).toHaveBeenCalled()
  })

  it('should render imageUrl as a prop on BackgroundImage', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(mockItem.imageUrl)
  })

  it('should render name prop in NameContainer', () => {
    expect(wrapper.find('NameContainer').text()).toBe(mockItem.name)
  })

  it('should render price prop in PriceContainer', () => {
    const price = parseInt(wrapper.find('PriceContainer').text())
    expect(price).toBe(mockItem.price)
  })
})
