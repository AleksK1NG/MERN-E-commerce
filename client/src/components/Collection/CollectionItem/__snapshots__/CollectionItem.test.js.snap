// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CollectionItem component should render CollectionItem component 1`] = `ShallowWrapper {}`;
